proposals/lip-0051.md

```diff
@@ -2246,29 +2246,17 @@ def removeAllTokensSupportFromChainID(chainID: ChainID) -> None:
 
 ```python
 def supportTokenID(tokenID: TokenID) -> None:
-    chainID = getChainID(tokenID)
-    # All tokens are supported.
-    if there exists entry in the supported tokens substore with key == ALL_SUPPORTED_TOKENS_KEY:
-        return
-
-    # Token already supported.
-    if chainID = OWN_CHAIN_ID or tokenID = getTokenIDLSK():
-        return
-
-    # All tokens from chain are supported.
-    if supportedTokens(chainID) exists:
-        if supportedTokens(chainID).supportedTokenIDs == []:
-            return
-
-        add tokenID to supportedTokens(chainID).supportedTokenIDs, maintaining the array in lexicographical order
 
-    else:
-        create an entry in the supported tokens substore with
-        key = chainID
-        value = encode(
-            schema=supportedTokensSchema,
-            object={"supportedTokenIDs": [tokenID]}
-        )
+    if not isTokenSupported(tokenID):
+	    if supportedTokens(chainID).supportedTokenIDs exists:
```

Comment:
The usual check in several other similar places is `if supportedTokens(chainID) exists`, I wonder if it is possible that `supportedTokens(chainID)` exists but it does not have the `supportedTokenIDs` property. Then the code below would create another / rewrite the entry in the supported tokens substore with key `chainID`.

For rigorousness and random bug prevention I would add another clause:
```
if supportedTokens(chainID).supportedTokenIDs exists:
    ...
else if supportedTokens(chainID) exists:
    # should never happen
    supportedTokens(chainID).supportedTokenIDs = [tokenID]
else:
    ...
```

Reply:
I think here there is a typo. It should be 

`if supportedTokens(chainID) exists:`

Since if the entry exists, it will always have a `supportedTokenIDs` property..

Reply:
This also replies to Sergey's comment.